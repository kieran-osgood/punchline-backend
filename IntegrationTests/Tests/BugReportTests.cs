
using System.Text.Json;
using System.Threading.Tasks;
using GraphQL;
using GraphQL.Data;
using GraphQL.Entities.JokeReport;
using HotChocolate;
using HotChocolate.Execution;
using HotChocolate.Types.Relay;
using IntegrationTests.Helpers;
using Microsoft.Extensions.DependencyInjection;
using Snapshooter.Xunit;
using Xunit;

namespace IntegrationTests.Tests
{
    public class BugReportTests : IClassFixture<CustomWebApplicationFactory<Startup>>
    {
        [Fact]
        public async Task Add_Bug_Report_Commits()
        {
            var factory =
                new CustomWebApplicationFactory<Startup>().WithAuthentication(TestClaimsProvider.WithUserClaims());
            const string description = "This joke was despicable!!!!!!!";

            var query = 
$@"mutation AddBugReport {{
    addBugReport(input: {{
        description: {"\"" + description+ "\""}
    }}) {{
    bugReport {{
        id
        description
    }}
    errors {{
        code
        message
    }}
  }}
}}"; 

            var executor = await factory.Services.GetRequestExecutorAsync();
            var request = QueryRequestBuilder
                .New()
                .AddAuthorizedUser()
                .SetQuery(query)
                .Create();

            var result = await executor.ExecuteAsync(request);
            
            Assert.Null(result.Errors);
            
            (await result.ToJsonAsync()).MatchSnapshot();
        }
    }
}