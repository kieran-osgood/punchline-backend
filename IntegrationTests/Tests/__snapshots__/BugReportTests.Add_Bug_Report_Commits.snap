{
  "data": {
    "addBugReport": {
      "bugReport": {
        "id": "QnVnUmVwb3J0Cmkx",
        "description": "This joke was despicable!!!!!!!"
      },
      "errors": null
    }
  }
}
